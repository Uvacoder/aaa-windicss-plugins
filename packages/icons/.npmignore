index.ts
tsconfig.json
rollup.config.js