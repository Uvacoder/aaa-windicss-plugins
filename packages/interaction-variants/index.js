const plugin = require('windicss/plugin');

module.exports = plugin(function ({ addVariant, e }) {
  addVariant('group-focus-within', ({ parent }) => {
    return parent('.group:focus-within');
  });

  addVariant('group-focus-visible', ({ parent }) => {
    return parent('.group:focus-visible');
  });

  addVariant('group-active', ({ parent }) => {
    return parent('.group:active');
  });

  addVariant('group-visited', ({ parent }) => {
    return parent('.group:visited');
  });

  addVariant('group-disabled', ({ parent }) => {
    return parent('.group:disabled');
  });

  addVariant('hocus', ({ modifySelectors }) => {
    return modifySelectors(({ className }) => {
      return `.${className}:hover, .${className}:focus`;
    });
  })

  addVariant('group-hocus', ({ modifySelectors }) => {
    return modifySelectors(({ className }) => {
      return `.group:hover .${className}, .group:focus .${className}`;
    });
  })

  addVariant('can-hover', ({ atRule }) => {
    return atRule('@media (hover: hover)');
  });

  addVariant('no-hover', ({ atRule }) => {
    return atRule('@media (hover: none)');
  })
});