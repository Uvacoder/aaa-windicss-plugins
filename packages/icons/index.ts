import icons from './icons.json';
import plugin from 'windicss/plugin';
import { CSSParser } from 'windicss/utils/parser';

const sizes: { [key:string]: string } = {
  'sm': '1',
  'md': '1.5',
  'lg': '2',
  'xl': '2.5'
}

export default plugin(({ addDynamic }) => {
  addDynamic('icon', ({ Utility, Property }) => {
    const body = Utility.body;
    if (body in sizes) return Property('--tw-icon', sizes[body])
    if (body in icons) {
      return new CSSParser((icons as {[key:string]: string})[body]).parse().children;
    }
  });
});
