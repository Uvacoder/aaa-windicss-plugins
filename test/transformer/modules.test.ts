import { Processor } from 'windicss/lib';

describe('modules', () => {
  it('tailwindcss-animate', () => {

    const processor = new Processor({
      plugins: [
        require('../../packages/tailwindcss-animate.css')({
          settings: {
            animatedSpeed: 1000,
            heartBeatSpeed: 1000,
            hingeSpeed: 2000,
            bounceInSpeed: 750,
            bounceOutSpeed: 750,
            animationDelaySpeed: 1000
          },
        })
      ]
    });
    expect(processor.interpret('animate-fadeIn animate-ping animate-delay-2s animate-shakeY animate-animated').styleSheet.build()).toMatchSnapshot('css');
  });

  it('tailwindcss-question-mark', () => {
    const processor = new Processor({
      plugins: [
        require('../../packages/tailwindcss-question-mark'),
      ]
    });
    expect(processor.interpret('? bg-white').styleSheet.build()).toMatchSnapshot('css');
  });
})