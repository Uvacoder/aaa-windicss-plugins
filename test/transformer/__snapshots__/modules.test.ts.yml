transform / tailwindcss-animate / css / 0: |-
  @keyframes fadeIn {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @-webkit-keyframes fadeIn {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }
  @keyframes ping {
    0% {
      -webkit-transform: scale(1);
      transform: scale(1);
      opacity: 1;
    }
    75%, 100% {
      -webkit-transform: scale(2);
      transform: scale(2);
      opacity: 0;
    }
  }
  @-webkit-keyframes ping {
    0% {
      -webkit-transform: scale(1);
      transform: scale(1);
      opacity: 1;
    }
    75%, 100% {
      -webkit-transform: scale(2);
      transform: scale(2);
      opacity: 0;
    }
  }
  @keyframes shakeY {
    from, to {
      -webkit-transform: translate3d(0, 0, 0);
      transform: translate3d(0, 0, 0);
    }
    10%, 30%, 50%, 70%, 90% {
      -webkit-transform: translate3d(0, -10px, 0);
      transform: translate3d(0, -10px, 0);
    }
    20%, 40%, 60%, 80% {
      -webkit-transform: translate3d(0, 10px, 0);
      transform: translate3d(0, 10px, 0);
    }
  }
  @-webkit-keyframes shakeY {
    from, to {
      -webkit-transform: translate3d(0, 0, 0);
      transform: translate3d(0, 0, 0);
    }
    10%, 30%, 50%, 70%, 90% {
      -webkit-transform: translate3d(0, -10px, 0);
      transform: translate3d(0, -10px, 0);
    }
    20%, 40%, 60%, 80% {
      -webkit-transform: translate3d(0, 10px, 0);
      transform: translate3d(0, 10px, 0);
    }
  }
  .animate-fadeIn {
    animation-name: fadeIn;
  }
  .animate-ping {
    -webkit-animation: ping 1s cubic-bezier(0, 0, 0.2, 1) infinite;
    animation: ping 1s cubic-bezier(0, 0, 0.2, 1) infinite;
  }
  .animate-delay-2s {
    animation-delay: 2000ms;
  }
  .animate-shakeY {
    animation-name: shakeY;
  }
  .animate-animated {
    animation-duration: 1000ms;
    animation-fill-mode: both;
  }
